README.md

```diff
@@ -15,13 +15,94 @@ refer to the documentation below for how to do that.
 ### Build requirements
  - a C++11 compiler (clang/gcc 4.7)
  - gnumake
- - cmake
- - boost
+ - CMake
+ - Boost
  - ruby (2.0 and newer)
- - [leatherman][leatherman](0.3.5 or newer)
+ - [leatherman][leatherman] (0.3.5 or newer)
  - [cpp-pcp-client][cpp-pcp-client] (master)
 
-Build with make and make install
+Initial Setup
+-------------
+
+Note: Testing custom facts requires Ruby 1.9+ with libruby built as a dynamic library; that often implies development builds of Ruby.
+
+### Setup on Fedora 23
+
+The following will install all required tools and libraries:
+
+    yum install boost-devel openssl-devel gcc-c++ make wget tar cmake
+
+### Setup on Mac OSX El Capitan (homebrew)
+
+This assumes Clang is installed and the system OpenSSL libraries will be used.
+
+The following will install all required libraries:
+
+    brew install cmake boost
+
+### Setup on Ubuntu 15.10 (Trusty)
+
+The following will install most required tools and libraries:
+
+    apt-get install build-essential libboost-all-dev libssl-dev wget tar cmake
+
+### Setup on Windows
+
+[MinGW-w64][MinGW-w64] is used for full C++11 support, and [Chocolatey][Chocolatey] can be used to install. You should have at least 2GB of memory for compilation.
+
+* install [CMake][CMake-choco] & [7zip][7zip-choco]
+
+        choco install cmake 7zip.commandline
+
+* install [MinGW-w64][MinGW-w64-choco]
+
+        choco install mingw --params "/threads:win32"
+
+For the remaining tasks, build commands can be executed in the shell from Start > MinGW-w64 project > Run Terminal
+
+* select an install location for dependencies, such as C:\\tools or cmake\\release\\ext; we'll refer to it as $install
+
+* build [Boost][Boost-download]
+
+        .\bootstrap mingw
+        .\b2 toolset=gcc --build-type=minimal install --prefix=$install --with-program_options --with-system --with-filesystem --with-date_time --with-thread --with-regex --with-log --with-locale --with-chrono boost.locale.iconv=off
+
+In Powershell:
+
+    choco install cmake 7zip.commandline -y
+    choco install mingw --params "/threads:win32" -y
+    $env:PATH = "C:\tools\mingw64\bin;$env:PATH"
+    $install = "C:\tools"
+
+    (New-Object Net.WebClient).DownloadFile("https://downloads.sourceforge.net/boost/boost_1_54_0.7z", "$pwd/boost_1_54_0.7z")
+    7za x boost_1_54_0.7z
+    pushd boost_1_54_0
+    .\bootstrap mingw
+    .\b2 toolset=gcc --build-type=minimal install --prefix=$install --with-program_options --with-system --with-filesystem --with-date_time --with-thread --with-regex --with-log --with-locale --with-chrono boost.locale.iconv=off
+    popd
+
+Build
+-----
+
+* build & install [leatherman][leatherman]
+
+* build & install [cpp-pcp-client][cpp-pcp-client]
+
+* build the pxp-agent
+
+  a release build:
+
+      mkdir release
+      cd release
+      cmake ..
+      make
+
+  a debug build
+
+      mkdir debug
+      cd debug
+      cmake -DTEST_VIRTUAL=ON -DDEV_LOG_COLOR=ON ..
```

Comment:
See above. I think it's better to just list and comment the options instead of suggesting the usage for a debug build.
But no strong opinion, so I leave it up to you.